client/src/screens/CourseRegistration/CourseRegistration.js

```diff
@@ -90,7 +72,62 @@ class CourseRegistration extends Component {
     }));
   }
 
+  // Returns the semesters displayed in the drop down
+  findSemesters(startYear) {
+    const today = new Date();
+    const currentYear = today.getFullYear();
+    let year = startYear;
+    let semesters = [];
+    for (year; year <= currentYear; year++) {
+      if (year === startYear) {
+        semesters.push('Fall ' + year.toString());
+      } else if (year === currentYear) {
+        semesters.push('Winter ' + year.toString());
+        if (today.getMonth() > 8) {
+          semesters.push('Fall ' + year.toString());
+        }
+      } else {
+        semesters.push('Winter ' + year.toString());
+        semesters.push('Fall ' + year.toString());
+      }
+    }
+    return semesters;
+  }
+
+  // TODO: remove because the student data should be passed down from login
+  fetchUserData() {
+    return axios.get(
+      'http://localhost:3001/users/getStudentData?studentID=999999999'
+    );
+  }
+
+  fetchAllCourses() {
+    return axios.get('http://localhost:3001/courses/getAllCourses');
```

Comment:
really minor suggestion, consider using a global variable to store url.  This way, if url changed, you don't need to modify everywhere 
```
const url = 'http://localhost:3001/';
axios.get(`${url}courses/getAllCourses`);
```

Reply:
the base url will be separated in a another file and exported so that it can be used everywhere. i'm going to leave my get request like this for now because it's easier to change once that is implemented